app/src/main/kotlin/com/saltedge/authenticator/features/launcher/LauncherActivity.kt

```diff
@@ -25,48 +25,63 @@ import android.content.Intent
 import android.os.Bundle
 import android.os.Handler
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
 import com.saltedge.authenticator.R
 import com.saltedge.authenticator.app.KEY_DEEP_LINK
+import com.saltedge.authenticator.app.LAUNCHER_SPLASH_DURATION
 import com.saltedge.authenticator.features.launcher.di.LauncherModule
+import com.saltedge.authenticator.features.settings.mvvm.about.ViewModelEvent
 import com.saltedge.authenticator.model.realm.RealmManager
 import com.saltedge.authenticator.sdk.constants.KEY_AUTHORIZATION_ID
 import com.saltedge.authenticator.sdk.constants.KEY_CONNECTION_ID
 import com.saltedge.authenticator.tool.*
 import com.saltedge.authenticator.tool.secure.updateScreenshotLocking
-import kotlinx.android.synthetic.main.activity_launcher.*
 import javax.inject.Inject
 
-const val LAUNCHER_SPLASH_DURATION = 1500L
-
 class LauncherActivity : AppCompatActivity() {
 
-    @Inject
-    lateinit var presenter: LauncherPresenter
+    lateinit var viewModel: LauncherViewModel
+    @Inject lateinit var viewModelFactory: LauncherViewModelFactory
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         injectDependencies()
+        setupViewModel()
         this.updateScreenshotLocking()
         this.applyPreferenceLocale()
         this.registerNotificationChannels()
         setContentView(R.layout.activity_launcher)
     }
 
-    override fun onResume() {
-        super.onResume()
-        animateComponents()
-        if (!AppTools.isTestsSuite(this)) RealmManager.initRealm(context = this)
-        if (RealmManager.errorOccurred) showDbError() else proceedToNextScreen()
+    private fun injectDependencies() {
+        authenticatorApp?.appComponent?.addLauncherModule(LauncherModule())?.inject(this)
+    }
+
+    private fun setupViewModel() {
+        viewModel = ViewModelProviders
+            .of(this, viewModelFactory)
+            .get(LauncherViewModel::class.java)
+        lifecycle.addObserver(viewModel)
+
+        viewModel.noErrorOccurred.observe(this, Observer<ViewModelEvent<Boolean>> {
+            it?.getContentIfNotHandled()?.let {
+                proceedToNextScreen()
+            }
+        })
+        viewModel.errorOccurred.observe(this, Observer<ViewModelEvent<Boolean>> {
+            it?.getContentIfNotHandled()?.let {
+                showDbError()
+            }
+        })
     }
 
     private fun proceedToNextScreen() {
         Handler().postDelayed({ startActivityWithPreset() }, LAUNCHER_SPLASH_DURATION)
     }
 
     private fun startActivityWithPreset() {
-        presenter.setupApplication()
-
-        this.startActivity(Intent(this, presenter.getNextActivityClass())
+        this.startActivity(Intent(this, viewModel.getNextActivityClass())
             .putExtra(KEY_CONNECTION_ID, intent.getStringExtra(KEY_CONNECTION_ID))
```

Comment:
business logic (where to get data and where to put data) should be implemented in ViewModel

Reply:
Added TODO